src/PyCall.jl

```diff
@@ -440,6 +440,59 @@ end
 #########################################################################
 
 """
+    @with
+
+Mimics a Python 'with' statement. Usage:
+
+@with EXPR as VAR begin
+    BLOCK
+end
+
+or 
+
+@with EXPR begin
+    BLOCK
+end
+"""
+macro with(EXPR,as,VAR,BLOCK)
+    @assert as==:as "usage: @with EXPR as VAR BLOCK."
+    _with(EXPR,VAR,BLOCK)
+end
+macro with(EXPR,BLOCK)
+    _with(EXPR,nothing,BLOCK)
+end
+        
+function _with(EXPR,VAR,BLOCK)        
+    quote
+        mgr = $(esc(EXPR))
+        exit = pybuiltin("type")(mgr)[:__exit__]  # Not calling it yet
+        value = pybuiltin("type")(mgr)[:__enter__](mgr)
```

Comment:
We should probably define
```jl
pytypeof(o::PyObject) = PyObject(@pycheckn ccall(@pysym(:PyObject_Type), PyPtr, (PyPtr,), o))
```
for these kinds of queries.  And probably use `mgrT = pytypeof(mgr)` to avoid calling it twice.

Use `["__exit__"]` and `["__enter__"]` to circumvent the type-introspection process, since we know we want PyObjects here.

Probably there should be a way to specify the type of object you want for `value`, by e.g. `@pywith expr as var::T ...`.   `T` would default to `PyAny` if `VAR` is specified, or `PyObject` if `VAR==nothing` (since in that case we don't need to convert the `__enter__` result to a native Julia type).  The you would call `value = pycall(mgrT["__enter__"], T, mgr)` to get the desired value type.